shotover-proxy/src/transforms/cassandra/sink_cluster/routing_key.rs

```diff
@@ -0,0 +1,2498 @@
+    let _ = cursor.write(&[0]);
+}
+
+#[cfg(test)]
+mod test_token_aware_router {
+    use super::super::node_pool::NodePool;
+    use super::super::routing_key::calculate;
+    use crate::transforms::cassandra::sink_cluster::node::CassandraNode;
+    use cassandra_protocol::consistency::Consistency::One;
+    use cassandra_protocol::frame::message_execute::BodyReqExecuteOwned;
+    use cassandra_protocol::frame::message_result::PreparedMetadata;
+    use cassandra_protocol::frame::message_result::{
+        ColSpec, ColType::Int, ColTypeOption, TableSpec,
+    };
+    use cassandra_protocol::frame::Version;
+    use cassandra_protocol::query::QueryParams;
+    use cassandra_protocol::query::QueryValues::SimpleValues;
+    use cassandra_protocol::token::Murmur3Token;
+    use cassandra_protocol::types::value::Value;
+    use cassandra_protocol::types::CBytesShort;
+    use uuid::uuid;
+
+    #[test]
+    fn test_router() {
+        let mut router = NodePool::new(prepare_nodes());
+
+        let prepared_metadata = PreparedMetadata {
+            pk_indexes: vec![0],
+            global_table_spec: Some(TableSpec {
+                ks_name: "test_ks".into(),
+                table_name: "my_test_table".into(),
+            }),
+            col_specs: vec![ColSpec {
+                table_spec: None,
+                name: "key".into(),
+                col_type: ColTypeOption {
+                    id: Int,
+                    value: None,
+                },
+            }],
+        };
+
+        let id = CBytesShort::new(vec![
+            11, 241, 38, 11, 140, 72, 217, 34, 214, 128, 175, 241, 151, 73, 197, 227,
+        ]);
+
+        let query_parameters = QueryParams {
+            consistency: One,
+            with_names: false,
+            values: Some(SimpleValues(vec![Value::Some(vec![0, 0, 1, 164])])),
+            page_size: None,
+            paging_state: None,
+            serial_consistency: None,
+            timestamp: None,
+            keyspace: None,
+            now_in_seconds: None,
+        };
+
+        let execute_body = &mut BodyReqExecuteOwned {
+            id: CBytesShort::new(vec![
+                11, 241, 38, 11, 140, 72, 217, 34, 214, 128, 175, 241, 151, 73, 197, 227,
+            ]),
+            result_metadata_id: None,
+            query_parameters: query_parameters.clone(),
+        };
+
+        router.add_prepared_result(id, prepared_metadata.clone());
+
+        let routing_key = calculate(
+            &prepared_metadata.pk_indexes,
+            query_parameters.values.as_ref().unwrap(),
+            Version::V4,
+        )
+        .unwrap();
+
+        assert_eq!(routing_key, vec![0, 0, 0, 4, 0, 0, 1, 164]);
+
+        let node = router.replica_node(execute_body, &Version::V4).unwrap();
+
+        assert_eq!(node.address, "172.16.1.5:9044".parse().unwrap());
+    }
+
+    fn prepare_nodes() -> Vec<CassandraNode> {
+        vec![
+            CassandraNode::new(
```

Comment:
Can we either reduce the size of the test data or move it into a separate file?

Reply:
Yeah ok, looking at how we actually use this test data, I really dont think its justified being so long, ~50 lines should be sufficient for the kind of test we are actually using it for.

Reply:
(I'm assuming you wrote this yourself and its not copied from cdrs-tokio right?)

Reply:
I just used a real token list from Cassandra

Reply:
Its ok for a unittest to use synthetic data, we have integration tests to cover more realistic cases.